src/treelearner/cost_effective_gradient_boosting.hpp

```diff
@@ -20,6 +20,8 @@ namespace LightGBM {
 
 class CostEfficientGradientBoosting {
  public:
+  std::vector<SplitInfo> splits_per_leaf_;
```

Comment:
why change this to public?

Reply:
I need to access it in my function UpdateBestSplitsFromHistograms in monotone_constraints.cpp. Should I write a getter instead to make it cleaner?

Reply:
yeah, setter and getter would be better.